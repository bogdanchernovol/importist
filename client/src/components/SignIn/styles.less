.title{
    margin-bottom: 31px;
}
.group-block{
    min-height: 340px;
    padding: 48px 30px 20px 30px;
}