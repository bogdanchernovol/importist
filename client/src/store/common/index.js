export {commonActions} from './actions';
export {CommonReducer} from './reducer';