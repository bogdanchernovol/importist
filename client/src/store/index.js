import "regenerator-runtime/runtime";
import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import Raven from "raven-js";
import createRavenMiddleware from "raven-for-redux";
import rootReducer from './reducers';
import sagas from './sagas';

Raven.config(process.env.SENTRY_API_KEY).install();

export default function configureStore(initialState) {
  
  const sagaMiddleware = createSagaMiddleware();
  const setry = createRavenMiddleware(Raven);
  
  function _getMiddleware() {
    const middleware = [      
      sagaMiddleware, 
      setry
    ];

    return applyMiddleware(...middleware);
  }
  const composeEnhancers =
    typeof window === 'object' && process.env.NODE_ENV !== 'production' &&
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ : compose;

  const store = composeEnhancers(_getMiddleware())(createStore)(
    rootReducer,
    initialState,      
  );

  sagaMiddleware.run(sagas);
  
  if (module.hot) {
    module.hot.accept(rootReducer, () => {
      store.replaceReducer(rootReducer.default);
    });
  }
  
  return store;
}