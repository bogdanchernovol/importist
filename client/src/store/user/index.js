export { userAction } from './actions';
export { UserReducer } from './reducer';
