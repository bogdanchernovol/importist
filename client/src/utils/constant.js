export const routeSignIn = '/sign-in';
export const routeSignUp = '/sign-up';
export const routeForgotPass = '/forgot-pass';
export const routeProducts = '/products';
export const routeSignUpPass = '/sign-up-pass';
export const routeVerifyEmail = '/verify-email/:id/:verifyCode';
export const routeAcceptInvite = '/accept-invite/:code';
export const routeComplteRegistration = '/complete-registration';
export const routeResetPassword = '/reset-password/:id/:verifyCode';
export const routeMyAccount = '/my-account';
export const routeMyProfile = '/my-account#myProfile';

export const ERROR = 'error';
export const SUCCESS= 'success';
export const PASS_RESETED = 'password reseted';

export const SUPPLIER = 'supplier';
export const CLIENT = 'client';
export const USER = 'user';
export const REGISTRATION_START = 'start';
export const REGISTRATION_SEND_EMAIL = 'send email';
export const REGISTRATION_EMAIL_VERIFIED = 'email verified';
export const REGISTRATION_COMPLETED = 'completed';

export const COUNT_STAFF = ['1-10', '11-50', '51-100', '100-500', '500+'];
export const REVENUE_AMOUNT = ['$0-1M', '$1M-$5M', '$5M-$20M', '$20M-$100M', '$100M-$1B', '$1B+'];
//TO-DO remove after user flow implementing
export const USER_PROFILE = {
    fullName: 'Andrew Jones',
    email: 'aj@supershippers.io',
    position: 'Lead Buyer',
    phone: '+1 620 485 2734',
}