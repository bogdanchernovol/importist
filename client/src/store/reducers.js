import { combineReducers } from 'redux';  
import {UserReducer} from './user';
import {CommonReducer} from './common';

export default  combineReducers({  
  userReducer: UserReducer,
  common: CommonReducer
});