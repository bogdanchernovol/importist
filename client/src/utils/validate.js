import * as constant from './constant';

export function validateEmail(email) { 
    // eslint-disable-next-line     
    const re = /^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i;
    if (email && email.match(re)) {
      return {
        validateStatus: constant.SUCCESS,
        errorMsg: null,
      };
    }
    return {
      validateStatus: constant.ERROR,
      errorMsg: 'Incorrect email.',
    };
}

export function  validateName(name) {      
    if (typeof name === 'string' && name.length > 2) {
        return {
          validateStatus: constant.SUCCESS,
          errorMsg: null,
        };
      }
      return {
        validateStatus: constant.ERROR,
        errorMsg: 'Incorrect full name.',
      };
}
export function  validatePassword(pass) {      
  if (typeof pass === 'string' && pass.length >= 8) {
      return {
        validateStatus: constant.SUCCESS,
        errorMsg: null,
      };
    }
    return {
      validateStatus: constant.ERROR,
      errorMsg: 'Incorrect password',
    };
}
export function validateConfirmPassword(confirmPass, password) {      
  if (typeof confirmPass === 'string' && confirmPass.length >= 8 && 
      confirmPass === password) {
      return {
        validateStatus: constant.SUCCESS,
        errorMsg: null,
      };
    }
    return {
      validateStatus: constant.ERROR,
      errorMsg: 'Incorrect confirm password',
    };
}

export function validateRequired(value) {      
  if (typeof value === 'string' && value.length >= 1) {
      return {
        validateStatus: constant.SUCCESS,
        errorMsg: null,
      };
    }
    return {
      validateStatus: constant.ERROR,
      errorMsg: 'Incorrect field value',
    };
}
export function validatePhone(phone) {
  const re = /^(\+\d{1,2}\s)?\(?\d{3}\)?[\s.-]?\d{3}[\s.-]?\d{4}$/;
  if (phone && phone.match(re)) {
    return {
      validateStatus: constant.SUCCESS,
      errorMsg: null,
    };
  }
  return {
    validateStatus: constant.ERROR,
    errorMsg: 'The Phone must be correct!',
  };
}