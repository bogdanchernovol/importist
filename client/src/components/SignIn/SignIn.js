import React, { Component } from 'react';
import {withRouter} from 'react-router';
import {connect} from 'react-redux';
import { Link } from 'react-router-dom';
import {userAction} from '../../store/user';
import {getUser} from '../../store/user/selectors';
import {Col, Row, Button} from '../../utils/antd';
import LoginForm from '../LoginForm';
import Auth from '../../containers/Auth';
import {validatePassword, validateEmail} from '../../utils/validate';
import * as constant from '../../utils/constant';
import './styles.less';

export class SignIn extends Component {
    constructor(props){
        super(props);
        this.state = this.initState();
    }
    initState = () =>{
        return {
            email: {},
            password: {}
        }
    }
    componentWillReceiveProps(nextProps){
        if (nextProps.user && this.props.user !== nextProps.user){
            this.props.history.push(constant.routeProducts);
        }
        if (nextProps.error){
            const field = nextProps.error.split(' ');
            if (field[field.length-1] === 'password'){
                this.setState({
                    password: {
                        validateStatus: constant.ERROR,
                        errorMsg: nextProps.error,
                        value: this.state.password.value
                    }
                }); 
            }
            if (field[field.length-1] === 'email'){
                this.setState({
                    email: {
                        validateStatus: constant.ERROR,
                        errorMsg: nextProps.error,
                        value: this.state.email.value
                    }
                });  
            }
        }
    }
        
    handleChangeEmail = (value)=>{        
        this.setState({
            email: {
                ...validateEmail(value),
                value
            }
        });
    }
    handleChangePass = (value)=>{        
        this.setState({
            password: {
                ...validatePassword(value, this.props.flag),
                value: value
            }
        });       
    }   
    onSignIn = ()=>{ 
        const email =  { ...validateEmail(this.state.email.value),  value: this.state.email.value};
        const password =  { ...validatePassword(this.state.password.value),  value: this.state.password.value};   
        if (email.validateStatus === constant.ERROR || password.validateStatus === constant.ERROR)  {
            this.setState({email, password});
            return;
        }   
        this.props.dispatch(userAction.clearError());
        this.props.dispatch(userAction.signIn({
            email: email.value,
            password: password.value
        }));        
    }
    componentWillUnmount(){
        this.props.dispatch(userAction.clearError());
    }
    render() { 
        const {email, password} = this.state;
         return (
            <Row  className='page-signIn display-all-center'>
                <Auth>
                    <Row className='group-block'>
                        <Row className='f-block'>
                            <Col span={24} className='title'>
                                Sign In
                            </Col>
                            <Col span={24}>
                                <LoginForm
                                    password={password}
                                    email={email}
                                    handleChangePass={this.handleChangePass}
                                    handleChangeEmail={this.handleChangeEmail}
                                />
                                <Col span={24} className={'btn-block'}>                                
                                    <Button 
                                        type="primary" 
                                        className='btn-green'
                                        onClick={this.onSignIn}>
                                        Sign In
                                    </Button>
                                </Col>
                                <Col span={24}>
                                    <Link to={constant.routeSignUp}>
                                        <Row className='btn-link-black'>
                                            Create Account
                                        </Row>                                    
                                    </Link>
                                </Col>  
                            </Col> 
                       </Row>
                    </Row>
                </Auth> 
                <Col span={24} className='forgot-link'>
                    <Link to={constant.routeForgotPass}>
                        Forgot your password?
                    </Link>
                </Col>                
            </Row>
        );
    }
}
function mapStateToProps(state){
    return {
        user: getUser(state), 
        error: state.userReducer.error
    }
}
export default withRouter(connect(mapStateToProps)(SignIn));