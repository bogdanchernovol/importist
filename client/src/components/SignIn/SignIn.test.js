import React from 'react';
import * as enzyme from 'enzyme';
const { shallow } = enzyme;
import { shallowToJson } from 'enzyme-to-json';
import ReactSixteenAdapter from 'enzyme-adapter-react-16';
import {validatePassword, validateEmail} from '../../utils/validate';
import {userAction} from '../../store/user';
import * as constant from '../../utils/constant';
import { SignIn } from './SignIn';

enzyme.configure({ adapter: new ReactSixteenAdapter() });
const dispatch = jest.fn();
const history = jest.fn()
history.push = jest.fn();

function setup() {
    const props = {
        error: '',
        user: {},
        dispatch,
        history
    };
    const renderedComponent = shallow(<SignIn {...props} />);
    return {
      props,
      renderedComponent,
    };
  }

  describe('SignIn Component', () => {
    it('should render self', () => {
        const { renderedComponent } = setup();
        expect(shallowToJson(renderedComponent)).toMatchSnapshot();
    });
    it('should change pass', () => {
        const { renderedComponent } = setup();
        const handleChangePass = renderedComponent.instance().handleChangePass;
        const value = '11111111';
        handleChangePass(value);
        expect(renderedComponent.state().password).toEqual({
                ...validatePassword(value),
                value: value            
        });
    });
    it('should change email', () => {
        const { renderedComponent } = setup();
        const handleChangeEmail = renderedComponent.instance().handleChangeEmail;
        const value = 'as@mail.com';
        handleChangeEmail(value);
        expect(renderedComponent.state().email).toEqual({
                ...validateEmail(value),
                value: value            
        });
    });
    it('should click onSignIn', () => {
        const { renderedComponent } = setup();
        const btn = renderedComponent.find('.btn-green');
        const email = 'as@mail.com';
        const password = '11111111';
        renderedComponent.state().email = {
            ...validateEmail(email),
                value: email
        }
        renderedComponent.state().password = {
            ...validatePassword(password),
                value: password
        }
        btn.simulate('click');
        expect(dispatch).toHaveBeenCalledWith(userAction.clearError());
        expect(dispatch).toHaveBeenCalledWith(userAction.signIn({
            email: renderedComponent.state().email.value,
            password: renderedComponent.state().password.value
        }));
    });
    it('should check initState', () => {
        const { renderedComponent } = setup();
        const initState = renderedComponent.instance().initState;
        initState();
        expect(renderedComponent.state()).toEqual({
                email: {},
                password: {}
        });
    });
    it('should change props - Incorrect password', () => {
        const { renderedComponent } = setup();
        const password = '11111111';
        const error =  'Incorrect password';
        renderedComponent.state().password = {
            ...validatePassword(password),
                value: password
        }
        // Change props
        renderedComponent.setProps({error});
        renderedComponent.state().password = {
            validateStatus: constant.ERROR,
            errorMsg: error,
            value: password
        }
    });
    it('should change props - Incorrect email', () => {
        const { renderedComponent } = setup();
        const email = 'as@mail.com';
        const error =  'Incorrect email';
        renderedComponent.state().email = {
            ...validateEmail(email),
                value: email
        }
        // Change props
        renderedComponent.setProps({error});
        renderedComponent.state().email = {
            validateStatus: constant.ERROR,
            errorMsg: error,
            value: email
        }
    });
    it('should change props - correct user', () => {
        const { renderedComponent, props } = setup();       
        renderedComponent.setProps({user: constant.USER_PROFILE});
        expect(props.history.push.mock.calls.length).toBe(1);
    });
  })


