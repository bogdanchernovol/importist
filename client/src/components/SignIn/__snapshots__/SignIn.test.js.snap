// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Component should render self 1`] = `
<Row
  className="page-signIn display-all-center"
  gutter={0}
>
  <Auth>
    <Row
      className="group-block"
      gutter={0}
    >
      <Row
        className="f-block"
        gutter={0}
      >
        <Col
          className="title"
          span={24}
        >
          Sign In
        </Col>
        <Col
          span={24}
        >
          <LoginForm
            email={Object {}}
            handleChangeEmail={[Function]}
            handleChangePass={[Function]}
            password={Object {}}
          />
          <Col
            className="btn-block"
            span={24}
          >
            <Button
              className="btn-green"
              ghost={false}
              loading={false}
              onClick={[Function]}
              prefixCls="ant-btn"
              type="primary"
            >
              Sign In
            </Button>
          </Col>
          <Col
            span={24}
          >
            <Link
              replace={false}
              to="/sign-up"
            >
              <Row
                className="btn-link-black"
                gutter={0}
              >
                Create Account
              </Row>
            </Link>
          </Col>
        </Col>
      </Row>
    </Row>
  </Auth>
  <Col
    className="forgot-link"
    span={24}
  >
    <Link
      replace={false}
      to="/forgot-pass"
    >
      Forgot your password?
    </Link>
  </Col>
</Row>
`;
